test/destructuring.jl

```diff
@@ -36,6 +36,17 @@ let
 end
 """) == ([1, 2, 3], 4)
 
+@test_throws JuliaLowering.LoweringError JuliaLowering.include_string(test_mod, """
+let
+    (x,) = ()
+end
+""") == ([1, 2, 3], 4)
+@test_throws JuliaLowering.LoweringError JuliaLowering.include_string(test_mod, """
+let
+    (x,y,ys...) = (1,)
+end
+""")
```

Comment:
Looks good, let's move both of these test cases into destructuring_ir.jl.  The IR tests have good infrastructure for pretty printing errors which make such tests nice to read and write. For example:

https://github.com/c42f/JuliaLowering.jl/blob/cfa6901d597dc27a303c28cc87fed608eb76df2a/test/destructuring_ir.jl#L85